ureg/src/lib.rs

```diff
@@ -32,24 +32,28 @@ mod private {
 
 impl Uint for u8 {
     const TYPE: UintType = UintType::U8;
+    fn from_u32(val: u32) -> Self {
+        (val & 0xff) as u8
+    }
 }
 impl private::Sealed for u8 {}
 
 impl Uint for u16 {
     const TYPE: UintType = UintType::U16;
+    fn from_u32(val: u32) -> Self {
+        (val & 0xffff) as u16
+    }
 }
 impl private::Sealed for u16 {}
 
 impl Uint for u32 {
     const TYPE: UintType = UintType::U32;
+    fn from_u32(val: u32) -> Self {
+        val
+    }
 }
 impl private::Sealed for u32 {}
 
-impl Uint for u64 {
```

Comment:
Why is u64 removed here?

Reply:
We never use 64-bit registers, and I don't know if we have the semantics for it super well-defined given our 32-bit memory bus. (Things like, which word is written first.) So, it seemed easier to just to remove 64-bit support altogether.

(Side note, I'd also be a fan of removing 8- and 16-bit register support as well, since it also is a little murky semantically, and refactoring out the `Uint` type altogether to just be `u32`.)

Reply:
I added it back in.